generator/standalone_main.cc

```diff
@@ -161,66 +174,52 @@ int WriteFeatureList(
   return 0;
 }
 
-}  // namespace
+google::protobuf::RepeatedPtrField<
+    google::cloud::cpp::generator::ServiceConfiguration>
+GenerateProtosForRestProducts(
+    CommandLineArgs const& generator_args,
+    google::protobuf::RepeatedPtrField<
+        google::cloud::cpp::generator::DiscoveryDocumentDefinedProduct> const&
+        rest_products) {
+  google::protobuf::RepeatedPtrField<
+      google::cloud::cpp::generator::ServiceConfiguration>
+      services;
 
-/**
- * C++ microgenerator.
- *
- * @par Command line arguments:
- *  --config-file=<path> REQUIRED should be a textproto file for
- *      GeneratorConfiguration message.
- *  --protobuf_proto_path=<path> REQUIRED path to .proto files distributed with
- *      protobuf.
- *  --googleapis_proto_path=<path> REQUIRED path to .proto files distributed
- *      with googleapis repo.
- *  --output_path=<path> OPTIONAL defaults to current directory.
- */
-int main(int argc, char** argv) {
-  absl::ParseCommandLine(argc, argv);
-  google::cloud::LogSink::EnableStdClog(
-      google::cloud::Severity::GCP_LS_WARNING);
-
-  auto proto_path = absl::GetFlag(FLAGS_protobuf_proto_path);
-  auto googleapis_path = absl::GetFlag(FLAGS_googleapis_proto_path);
-  auto golden_path = absl::GetFlag(FLAGS_golden_proto_path);
-  auto config_file = absl::GetFlag(FLAGS_config_file);
-  auto output_path = absl::GetFlag(FLAGS_output_path);
-  auto scaffold = absl::GetFlag(FLAGS_scaffold);
-  auto scaffold_templates_path = absl::GetFlag(FLAGS_scaffold_templates_path);
-  auto experimental_scaffold = absl::GetFlag(FLAGS_experimental_scaffold);
-
-  GCP_LOG(INFO) << "proto_path = " << proto_path << "\n";
-  GCP_LOG(INFO) << "googleapis_path = " << googleapis_path << "\n";
-  GCP_LOG(INFO) << "config_file = " << config_file << "\n";
-  GCP_LOG(INFO) << "output_path = " << output_path << "\n";
-
-  auto config = GetConfig(config_file);
-  if (!config.ok()) {
-    GCP_LOG(ERROR) << "Failed to parse config file: " << config_file << "\n";
+  for (auto const& p : rest_products) {
+    auto status =
+        google::cloud::generator_internal::GenerateProtosFromDiscoveryDoc(
+            p.discovery_document_url(), generator_args.protobuf_proto_path,
+            generator_args.googleapis_proto_path, generator_args.output_path);
+    if (status.ok()) {
```

Comment:
Are we presuming that `GenerateProtosFromDiscoveryDoc()` has already emitted an error message if it returns a bad `Status`?  I think I'd make this be the error-message emitter instead.